hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/cdc/HoodieCDCRDD.scala

```diff
@@ -157,7 +154,7 @@ class HoodieCDCRDD(
         split.changes.last.getInstant,
         recordKeyField,
         preCombineFieldOpt,
-        usesVirtualKeys = false,
+        usesVirtualKeys = !populateMetaFields,
```

Comment:
this is a fix among those renaming changes

Reply:
Seems critical, should be merged for RC2

Reply:
Restore this first. let's keep this pr force on code-improvement.

Reply:
separate PR for the fix?

Reply:
https://github.com/apache/hudi/pull/7777